### PR TITLE
Add element_at for map

### PR DESCRIPTION
**Pull request by cberner**
This is in preparation for changing the behavior of map subscript to
throw an exception when the key is not found.

### CODE REVIEW

**Comment by cberner** [comment]
The implementation is the same as map subscript operator

**Comment by cberner** [comment]
@raghavsethi I discussed the signature of this already with @martint, so can you just look at the implementation?

File: presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java [hunk 1]
```diff
@@ -206,6 +206,26 @@ public void testJsonToMap()
...
+        assertFunction("element_at(MAP(ARRAY [1, 3], ARRAY [2, 4]), 3)", INTEGER, 4);
+        assertFunction("element_at(MAP(ARRAY [BIGINT '1', 3], ARRAY [BIGINT '2', 4]), 3)", BIGINT, 4L);
+        assertFunction("element_at(MAP(ARRAY [1, 3], ARRAY[2, NULL]), 3)", INTEGER, null);
+        assertFunction("element_at(MAP(ARRAY [BIGINT '1', 3], ARRAY[2, NULL]), 3)", INTEGER, null);
+        assertFunction("element_at(MAP(ARRAY [1, 3], ARRAY [2.0, 4.0]), 1)", DOUBLE, 2.0);
+        assertFunction("element_at(MAP(ARRAY[1.0, 2.0], ARRAY[ ARRAY[1, 2], ARRAY[3]]), 1.0)", new ArrayType(INTEGER), ImmutableList.of(1, 2));
+        assertFunction("element_at(MAP(ARRAY['puppies'], ARRAY['kittens']), 'puppies')", createVarcharType(7), "kittens");
+        assertFunction("element_at(MAP(ARRAY[TRUE,FALSE],ARRAY[2,4]), TRUE)", INTEGER, 2);
+        assertFunction("element_at(MAP(ARRAY['1', '100'], ARRAY[from_unixtime(1), from_unixtime(100)]), '1')", TIMESTAMP, new SqlTimestamp(1000, TEST_SESSION.getTimeZoneKey()));
+        assertFunction("element_at(MAP(ARRAY[from_unixtime(1), from_unixtime(100)], ARRAY[1.0, 100.0]), from_unixtime(1))", DOUBLE, 1.0);
```
Comment (raghavsethi) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: What about testing for elements not present in the map? Also maybe add a test for an empty map?

**Comment by cberner** [comment]
Updated

**Comment by raghavsethi** [comment]
Looks good

**Pull request merged**